sympy/polys/domains/powerseriesring.py

```diff
@@ -0,0 +1,126 @@
+from sympy.core.expr import Expr
+from sympy.polys.series.ring import PowerSeriesRing, PowerSeriesElement, TSeries
+from sympy.polys.domains.ring import Ring
+from sympy.polys.domains.compositedomain import CompositeDomain
+from sympy.polys.domains.domain import Er, Domain
+from sympy.utilities import public
+
+from typing import Union
+
+@public
+class SeriesRing(Ring[PowerSeriesElement[Er]], CompositeDomain):
+    """A Domain class for representing univariate power series rings."""
+
+    is_PowerSeriesRing = is_Series = True
+
+    has_assoc_Ring = True
+    has_assoc_Field = False
+
+    ring: PowerSeriesRing[Er]
+    dtype: type[PowerSeriesElement[Er]]
+    gen: PowerSeriesElement[Er]
+    symbol: Expr
+    domain: Domain[Er]
+
+    def __init__(self, domain: Domain[Er], symbol = "x", prec: int = 6):
+        ring = PowerSeriesRing(domain, symbol, prec)
+
+        self.ring = ring
+        self.dtype = ring.dtype
+
+        self.domain = ring.domain
+        self.gen = ring.gen
+        self.symbol = ring.symbol
+
+    def new(self, element: Union[TSeries, Expr, Er, int]) -> PowerSeriesElement[Er]:  # type: ignore
+        return self.ring.ring_new(element)
+
+    def of_type(self, element) -> bool:
+        """Check if ``a`` is of type ``dtype``."""
+        return self.ring.is_element(element)
+
+    @property
+    def zero(self) -> PowerSeriesElement[Er]:  # type: ignore
+        return self.ring.zero
+
+    @property
+    def one(self) -> PowerSeriesElement[Er]:  # type: ignore
+        return self.ring.one
+
+    @property
+    def prec(self) -> int:
+        return self.ring.prec
+
+    def __repr__(self) -> str:
+        return f"{self.domain}[[{self.symbol}], {self.prec}]"
+
+    def __hash__(self) -> int:
+        return hash((self.__class__.__name__, self.ring, self.domain, self.symbol))
+
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, SeriesRing):
+            return NotImplemented
+        return self.ring == other.ring
+
+    def is_unit(self, a: PowerSeriesElement[Er]) -> bool:
+        """Returns ``True`` if ``LC`` of series is a unit of ``self``"""
+        if not a.is_ground:
+            return False
+        K = self.domain
+        return K.is_unit(K.convert_from(a, self))
+
+    def canonical_unit(self, a: PowerSeriesElement[Er]) -> PowerSeriesElement[Er]:
+        u = self.domain.canonical_unit(a.leading_coefficient())
+        return self.ring.ground_new(u)
+
+    def to_sympy(self, a: PowerSeriesElement[Er]) -> Expr:
+        """Convert `a` to a SymPy object."""
+        return a.as_expr()
+
+    def from_sympy(self, a: Expr) -> PowerSeriesElement[Er]:
+        """Convert SymPy's expression to `dtype`."""
+        return self.ring.from_expr(a)
+
+    def from_ZZ(K1, a, K0):
+        """Convert a Python `int` object to `dtype`."""
+        return K1(K1.domain.convert(a, K0))
```

Comment:
I think it is generally better if these methods don't fallback on calling `convert` but rather call some specific function that handles the ZZ case. It might be necessary to add such methods to the ring.